### ISSUE TITLE
limit from variable

### ISSUE THREAD

**Issue by antromanfdez** [ansible/awx#12075]
##### ISSUE TYPE
 - Feature Idea

##### SUMMARY

I would like to be able to provide a variable instead of a pattern for **limit**.

The purpose is to be able to get through set_stat in step 1 of
a workflow those hosts that I do not want to run in step 2 of this workflow.

**I **am getting the following warning****:
**INFO**: ansible_limit contains in this case the require host for testing purposes '192..168.56.20'

`[WARNING]: Could not match supplied host pattern, ignoring: ansible_limit
ERROR! Specified hosts and/or --limit does not match any hosts`

**Comment by AlanCoding** [comment]
Could you give an example of what you want to do using the `ansible-playbook` CLI? You can see some examples of `--limit` syntax here https://docs.ansible.com/ansible/latest/user_guide/intro_patterns.html

If there's anything you can do with that which you can't do in AWX, then it's clearly something we should support. I don't know it really supports jinja2 templating or host expansion patterns, but I can't rule it out.

**Comment by antromanfdez** [comment]
In the following workflow I am using three templates:
![workflow](https://user-images.githubusercontent.com/30831001/166271676-49fb55b8-5149-4bf2-9357-8e6b7cc2aaef.JPG)

 **testing_login:** 
 In this template I am collecting information of the different host and append to a set_stat variable named `collect_info`

**wall_failed:**
this template that will only be executed from awx in localhost mode will analyze the collected data with the `collect_info `variable and create another variable with set_stat called `exclude_failed`

The following image shows the detail of the execution of the **wall_failed** template and the hosts that I want to exclude in the next execution of the workflow.

![image](https://user-images.githubusercontent.com/30831001/166292848-3421fde5-1a3d-46c6-9adc-30d2656a1c47.png)


**testing_url**

In this playbook I would like to be able to use the `exclude_failed` variable to exclude these hosts from execution for this particular template.

The following image shows the detail of the `exclude_failed `variable of the **testing_url** template

![image](https://user-images.githubusercontent.com/30831001/166295526-ff1729f7-2a62-4930-9fce-f803b9087cea.png)


Is there a way to use this variable in `limit `to exclude these hosts?

 Is it necessary to use `extra-vars` in the template itself to somehow tell `limit `to exclude these hosts?

**Comment by AlanCoding** [comment]
That's true that AWX doesn't have any mechanism to support this. I get the use case now, but I don't have a design in mind.

**Comment by antromanfdez** [comment]
Thank you very much in advance @AlanCoding  for your support.  I hope that this enhancement can be considered for inclusion in awx one day.

**Comment by unbreakabl3** [comment]
I will vote for that feature as well. just got stuck with that now :(
